src/Castle.Windsor/MicroKernel/Releasers/LifecycledComponentsReleasePolicy.cs

```diff
@@ -147,7 +164,10 @@ public void Release(object instance)
 			}
 
 			Burden burden;
-			using (@lock.ForWriting())
+
+			@lock.EnterWriteLock();
```

Comment:
As a 1:1 conversion, this looks good. But I can't help feeling a bit confused about what is going on here. This code is taking a write lock when it's only reading a dictionary. The `Release` call (which I assume triggers said `OnInstanceReleased` event?) happens outside the lock. Should this probably be an upgradeable read lock, with the call to `Release` being moved inside it... or am I misunderstanding how this code works?